test/unit/collections/base.js

```diff
@@ -146,4 +146,35 @@ describe('Unit: Collections: Base', function() {
 
   })
 
+  it('should be an iterable', function(done) {
+
+    class Collection extends Dragon.Collection {}
+    class Model      extends Dragon.Model {}
+
+    var collection = new Collection()
+    
+    collection.add([new Model({foo: 'bar'}), new Model({foobar : 'baz'}), new Model({foovar : 'varbaz'})])
+
+    /* now our collection can be iterable for iterators like for ...of, ... spread operators,array.from */
+    for(var singleModel of collection){
```

Comment:
@chrisabrams what do you think about this new feature ? i think is helpful instead of the user access to collection.models , the collection object is an iterable itself :bowtie:
FYI:  http://www.2ality.com/2015/02/es6-iteration.html